### PR TITLE
Service discovery bug fixes

### PR DESCRIPTION
**Pull request by alena1108**
1) Validate against ports when add/remove lbTarget

https://github.com/rancher/rancher/issues/1508

2) Allow setting empty list of services in setServiceLinks

3) Populate sidekick labels on config.export

https://github.com/rancher/rancher/issues/1362

### CODE REVIEW

**Comment by alena1108** [comment]
@sonchang @cjellick could you please review?

File: tests/integration/cattletest/core/test_svc_discovery_lb.py [hunk 1]
```diff
@@ -459,8 +459,9 @@ def test_labels(super_client, client, context):
                                                 service, client,
                                                 ['8089:8089', '914:914'])
     lb_instance = _validate_lb_instance(host, lb, super_client, service)
-    name = env.name + '/' + service_name + '/' + \
-        "io.rancher.service.primary.launch.config"
+    lc = "io.rancher.service.primary.launch.config"
+    name = env.name + '/' + service_name + '/' + lc
```
Comment (sonchang) [inline tests/integration/cattletest/core/test_svc_discovery_lb.py]: FYI: I've removed the primary launch config part of the service name

Comment (alena1108) [inline tests/integration/cattletest/core/test_svc_discovery_lb.py]: ok, I should get these changes with rebase

**Comment by sonchang** [comment]
LGTM

**Comment by cjellick** [comment]
@alena1108 I reviewed commit by commit, so I put a comment on a specific commit. ..Let me know if that is confusing.

**Comment by cjellick** [comment]
I may be naive here, but seems like it would be fairly straightforward to write a test for verifying that sidekick labels get populated. Is that so or is there some complexity that makes that not worth the effort?

**Comment by alena1108** [comment]
@cjellick  - I am yet to add an integration test for the entire exportConfig use case. So far it gets verified only in validation tests. Support for exported config will most likely come as a separate PR.

**Comment by cjellick** [comment]
LGTM
...assuming you dont want to make the StringBuffer to StringBuilder change (or want to make it later)
Cant merge, need to rebase :-(

**Pull request merged**